shell/common/shell.h

```diff
@@ -243,6 +243,12 @@ class Shell final : public PlatformView::Delegate,
   Rasterizer::Screenshot Screenshot(Rasterizer::ScreenshotType type,
                                     bool base64_encode);
 
+  /// Pauses the calling thread until the first frame is presented.
+  ///\details Don't call this from the GPU thread or the UI thread or you will
+  ///         create a deadlock.
+  ///\returns true when there has been a timeout.
+  bool WaitForFirstFrame(fml::TimeDelta timeout);
```

Comment:
In this case, the return type is hard to interpret (I initially expected true meant it waited). I suggest a enum return value.

```
enum class WaitResult {
 kWaited, // or, kSuccess or whatever
 kTimeout
};
```

Reply:
I tend to agree, but I was being consistent with the rest of our API: https://github.com/flutter/engine/blob/7f4f52f95294544a2970e514d163b9791efa9147/fml/synchronization/waitable_event.h#L58

Reply:
I've added a Status class as discuss offline.